lib/events/filesessions/fileasync.go

```diff
@@ -348,10 +365,24 @@ func (u *Uploader) startUpload(ctx context.Context, fileName string) error {
 		return trace.Wrap(err)
 	}
 	if isSessionError {
```

Comment:
Couple of questions here:

- Can we get the `isSessionError` for any other reason than corrupted file?
- Can it happen that the "corrupted" upload becomes "uncorrupted"? What I mean is, would it make sense to, say, declare it corrupted only after N unsuccessful upload attempts instead of right away?

Reply:
No, that error is only for corrupted files. If you look at its `Error() string` method you'll see it prefixes every error with "the file is corrupted".

This happens if the file contains data that can not be read (either because it's not properly gzipped, or because the uncompressed data inside is not a protobuf). In any case, the contents of the file on disk aren't going to magically change, so a retry is not going to have different results.

(This is why we're seeing customers with huge CPU spikes. Even though we don't try to re-upload the file after we know it's corrupted, we still call `startUpload` only to realize that we shoudl skip this upload over and over and over again)